README.md

```diff
@@ -114,7 +113,11 @@ jobs:
 
 ### Tokens
 
-Workflows completed using the `secrets.GITHUB_TOKEN` will not trigger other workflow actions, hence the use of `secrets.GU_GITHUB_TOKEN` for the step which opens the pull request. As merging the pull request does not need to trigger another step, the `secrets.GITHUB_TOKEN` can be used for this part.
+Workflows completed using the `secrets.GITHUB_TOKEN` will not trigger other workflow actions, hence the use of `secrets.GU_GITHUB_TOKEN` for the step which opens the pull request. As merging the pull request does not need to trigger another step, the `secrets.GITHUB_TOKEN` can be used for this part. See the `environments` section for [more information on how to configure secrets in your repository](#environments).
+
+### Environments
+
+[GitHub environments](https://docs.github.com/en/actions/reference/environments) are "spaces" that can be configured with protection rules and secrets. Workflows that reference an environment will be bound by its protection rules and, if allowed, get access to it's secrets. When configuring auto publishing, it is recommended that you create a new environment for this purpose, configured so that only the release branch can use it. You can then add the `GU_GITHUB_TOKEN` and `GU_NPM_TOKEN` as environment secrets and reference the environment in the `CD` workflow. This means that only workflows running on the main branch will have access to the secrets required for release. This approach should be used in conjuction with the [recommended repository settings](#repository-settings).
```

Comment:
as mentioned in chat this feels like really useful insight, esp with the repo setting below, that maybe we should capture as a recommendation. it could be a bit lost if it only ever lived here

Reply:
Agreed. I'll merge this for now and then we can figure out which bits of it we want to move over to recommendations.